### PR TITLE
Documentation Update: Primarily Ruby Docs

### PR DESCRIPTION
**Pull request by luke-hill**
Few different fixes (Mainly ruby), from OSS day.

Nothing really removed (Other than the re-using of browser windows)

### CODE REVIEW

File: content/browser-automation.md
Comment (mlvandijk) [inline content/browser-automation.md]: Afaik Serenity is java only, so I understand marking the whole thing. Do we provie an alternate text for other languages?

File: content/browser-automation.md
Comment (mlvandijk) [inline content/browser-automation.md]: This would be the alternate text for Ruby and javascript :) (see my previous comment)

File: content/browser-automation.md
Comment (mlvandijk) [inline content/browser-automation.md]: Similarly, add an alternate text to the ruby specific ones?

File: content/gherkin.md
Comment (mlvandijk) [inline content/gherkin.md]: lower case "Optional" -> "optional" (although it's implied by the rest of the sentence, both name and description are optional

File: content/gherkin.md
Comment (mlvandijk) [inline content/gherkin.md]: lower case "Keyword" -> "keyword" for consistency please

File: content/gherkin.md
Comment (mlvandijk) [inline content/gherkin.md]: Nice catch, thnx!

File: content/browser-automation.md
Comment (mlvandijk) [inline content/browser-automation.md]: Nice catch, thnx!

File: content/gherkin.md
Comment (mlvandijk) [inline content/gherkin.md]: "_should_" -> "*should*" for consistency please

File: content/gherkin.md
Comment (mlvandijk) [inline content/gherkin.md]: "for different features, then it's indicative of you needing to isolate your test cases into 2 or more features." ->
"for different scenarios, you'll need to split them into different feature files." (shorter/clearer)

File: content/gherkin.md
Comment (mlvandijk) [inline content/gherkin.md]: Nice!

**Review by mlvandijk — COMMENTED** [review]
Thanks for your work @luke-hill ! Some very small comments inline if you don't mind. And ofcourse I can't comment on your Ruby changes. But if someone else could, I'd be happy to merge. Thanks!

File: content/installation.md
Comment (diabolo) [inline content/installation.md]: No, its important that cucumber does not get into production. In a Gemfile it should always be in a non-production group

File: content/gherkin.md
Comment (diabolo) [inline content/gherkin.md]: I don't think we need the `a ` here. In this context state is a concept not a singular thing

File: content/gherkin.md
Comment (diabolo) [inline content/gherkin.md]: resist `that` temptation

File: content/gherkin.md
Comment (diabolo) [reply content/gherkin.md]: don't think you need optional here, prefer the original

File: content/browser-automation.md
Comment (diabolo) [reply content/browser-automation.md]: I think we should break these tool descriptions out into separate files e.g. `tools\java\serenity-bdd` and just have the link in here. Something for future work on this file

**Review by diabolo — CHANGES_REQUESTED** [review]
Hey, terrific stuff, thanks for all you work on this. 

I'm new to these docs, first review of a PR with them. Hope requested changes are useful :)

File: content/browser-automation.md
Comment (luke-hill) [reply content/browser-automation.md]: Agreed. The problem is the way the syntax works. You can't wrap syntax over `##` The build process complains. If you see my 3rd commit (You'll notice it failed the build), this was the reason why.

File: content/installation.md
Comment (luke-hill) [reply content/installation.md]: This is the 10-minute installation and is using ruby as a standalone tool. Not as part of a server. See below for the rails information

**Comment by luke-hill** [comment]
For reference for both of you have a quick glance through: https://app.netlify.com/sites/cucumber/deploys/5adb44c8e39e7c1137b34c59 - Evidently at the time I found something that doesn't play with the build process. Not sure what to do. Figured what I did was the quickest. But I don't like all the non-relevant titles in each thread.

Maybe one for @aslakhellesoy to advise on?

File: content/installation.md
Comment (diabolo) [reply content/installation.md]: Tutorials tend to set patterns that new users follow for a long time. Its better to have a bit more stuff for them to cut and paste rather than set a precedent that they probably will not have enough knowledge to unlearn for quite a while

File: content/browser-automation.md
Comment (mlvandijk) [reply content/browser-automation.md]: Yes, I am aware ;) I'll make a note and fix separately.

File: content/gherkin.md
Comment (mlvandijk) [reply content/gherkin.md]: Same :)

File: content/installation.md
Comment (mlvandijk) [reply content/installation.md]: So the original `group :test do` gives it a test scope? (non-Ruby user ;) I'd prefer to keep that then.

File: content/gherkin.md
Comment (mlvandijk) [reply content/gherkin.md]: Agree

File: content/installation.md
Comment (luke-hill) [reply content/installation.md]: It depends. It has no use in regular ruby environments. There's a few things in here that aren't strictly wrong that I've fixed (Such as rockets `=> are an old legacy syntax` e.t.c.)

For those old styles there are 2 trains of thought. There isn't really a "right" answer. In my eyes I wouldn't ever teach people about using grouping inside bundler unless they were doing Rails. It has no use and just muddies the file.

Likewise, if you teach them to use grouping. Then when it comes to Rails they don't need to learn what it does.

My opinion is that because it doesn't actually do anything, they might expect it to, which is wrong.

My 0.02c

File: content/browser-automation.md
Comment (mlvandijk) [reply content/browser-automation.md]: Still thinking that as the rest of the page is polyglot it would be ok to have the actual text for Serenity marked as "java" text blok, but have an alternate text for Ruby, javascript with something like "Serenity is jvm only" or something. And vice versa for the Ruby tools. (to prevent having too many in depth pages that would then loose context as @mattwynne mentioned in #151

**Comment by luke-hill** [comment]
As it seems as though there have been some sweeping changes. It's likely some of my stuff can now be removed (As it broke builds previously).

Fixing the conflict was rather tricky. But I've done it (And fixed a couple of the review points), a couple of them didn't make sense to me, so I've not fixed them. But feel free to re-raise if you feel appropriate.

I know I can look into the java tagging stuff in more detail. But I'd rather find out firstly what you think needs fixing before I sink a load more time in.

**Comment by mlvandijk** [comment]
@luke-hill One small edit for consistency. Otherwise, I'm to merge this (unless @diabolo objects?) and take it from there (I can have a look at the text blocks for browser automation (or it might be split into several languages specific pages).

**Review by mlvandijk — APPROVED** [review]

**Comment by aslakhellesoy** [comment]
Hi @luke-hill,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Comment by mattwynne** [comment]
Yay welcome @luke-hill!

**Pull request merged**
